packages/editor/src/client/panel/peerEditPanel.tsx

```diff
@@ -101,27 +103,31 @@ export function PeerEditPanel({id, onClose, onSave}: ImageEditPanelProps) {
           setLoadingPeerProfile(false)
 
           // TODO: Better validation
-          if (response?.data?.peerProfile) {
-            setValidURL(true)
-            setProfile(response.data.peerProfile)
+          if (!response?.data?.peerProfile) {
+            setValidPeer(false)
+            setErrorMessage(t('peerList.panels.invalidURL'))
+          } else if (response?.data?.peerProfile.hostURL !== hostURL) {
+            setValidPeer(false)
+            setErrorMessage(t('peerList.panels.peeredToHost'))
           } else {
-            setValidURL(false)
+            setValidPeer(true)
```

Comment:
i think here you need to set the `ErrorMessage` to `''` otherwise you'll have a valid peer but still show an error.